src/components/Microsoft.DotNet.UpgradeAssistant.Extensions/ExtensionProviderExtensions.cs

```diff
@@ -35,60 +36,63 @@ public static void AddExtensions(this IServiceCollection services, IConfiguratio
                 throw new ArgumentNullException(nameof(configuration));
             }
 
-            services.AddTransient<IUpgradeStartup, ExtensionLoggingStartup>();
-            services.AddSerializer();
-
-            foreach (var extension in GetExtensions(configuration, additionalExtensionPaths))
-            {
-                services.AddSingleton(extension);
-
-                foreach (var sp in extension.GetServiceProviders())
+            services.AddOptions<ExtensionOptions>()
+                .Configure(options =>
                 {
-                    sp.AddServices(new ExtensionServiceCollection(services, extension));
-                }
-            }
-        }
+                    foreach (var path in additionalExtensionPaths)
+                    {
+                        options.ExtensionPaths.Add(path);
+                    }
+                })
+                .AddDefaultExtensions(configuration)
```

Comment:
The order extensions are added matters (if two extensions add template files with the same path, the later one wins) so we need to make sure we're consistently registering extensions in the same order. To match old behavior, I think the order needs to be default extensions loaded first, then from env vars, then from command line.